examples/aws.yaml

```diff
@@ -1,215 +1,69 @@
-    #
-    # Example: `"Z1ILINNUJGTAO1"`
-    # privateZone:
-
-    # (optional) ID of an existing VPC to launch nodes into.
-    # If unset a new VPC is created.
-    #
-    # Example: `vpc-123456`
-    # vpcID:
-
-    # (optional) List of subnet IDs within an existing VPC to deploy worker nodes into.
-    # Required to use an existing VPC and the list must match the AZ count.
-    #
-    # Example: `["subnet-111111", "subnet-222222", "subnet-333333"]`
-    # workerSubnetIDs:
-
-  # (optional) Extra AWS tags to be applied to created resources.
-  #
-  # Example: `{ "key" = "value", "foo" = "bar" }`
-  # extraTags:
+metadata:
 
-  # (optional) Name of IAM role to use to access AWS in order to deploy the Tectonic Cluster.
-  # The name is also the full role's ARN.
+  # The name of the cluster.
+  # If used in a cloud-environment, this will be prepended to `baseDomain` to construct a domain for the OpenShift console.
   #
-  # Example:
-  #  * Role ARN  = arn:aws:iam::123456789012:role/tectonic-installer
-  # installerRole:
-
-  master:
-    # (optional) This configures master availability zones and their corresponding subnet CIDRs directly.
-    #
-    # Example:
-    # `{ eu-west-1a = "10.0.0.0/20", eu-west-1b = "10.0.16.0/20" }`
-    # customSubnets:
-
-    # Instance size for the master node(s). Example: `t2.medium`.
-    ec2Type: t2.medium
-
-    # (optional) List of additional security group IDs for master nodes.
-    #
-    # Example: `["sg-51530134", "sg-b253d7cc"]`
-    # extraSGIDs:
-
-    # (optional) Name of IAM role to use for the instance profiles of master nodes.
-    # The name is also the last part of a role's ARN.
-    #
-    # Example:
-    #  * Role ARN  = arn:aws:iam::123456789012:role/tectonic-installer
-    #  * Role Name = tectonic-installer
-    # iamRoleName:
-
-    rootVolume:
-      # The amount of provisioned IOPS for the root block device of master nodes.
-      # Ignored if the volume type is not io1.
-      iops: 100
-
-      # The size of the volume in gigabytes for the root block device of master nodes.
-      size: 30
-
-      # The type of volume for the root block device of master nodes.
-      type: gp2
-
-  # (optional) If set to true, create private-facing ingress resources (ELB, A-records).
-  # If set to false, no private-facing ingress resources will be provisioned and all DNS records will be created in the public Route53 zone.
-  # privateEndpoints: true
-
-  # (optional) This declares the AWS credentials profile to use.
-  # profile: default
-
-  # (optional) If set to true, create public-facing ingress resources (ELB, A-records).
-  # If set to false, no public-facing ingress resources will be created.
-  # publicEndpoints: true
-
-  # The target AWS region for the cluster.
-  region: us-east-1
+  # Note: This field MUST be set manually prior to creating the cluster.
+  # Warning: Special characters in the name like '.' may cause errors on OpenStack platforms due to resource name constraints.
+  name:
```

Comment:
I would like to see https://github.com/openshift/installer/pull/209 so that name is a key in config, rather than in metadata.

Reply:
> I would like to see #209...

We should probably have this discussion over in #209.  I'll rebase this PR to follow whatever's in master underneath.